ulid.go

```diff
@@ -454,3 +467,85 @@ func (id *ULID) Scan(src interface{}) error {
 func (id ULID) Value() (driver.Value, error) {
 	return id.MarshalBinary()
 }
+
+// Monotonic returns an entropy source that is guaranteed to return
+// strictly increasing entropy bytes for the same ULID timestamp.
+// On conflicts, the previous ULID entropy is incremented with a
+// random number between 1 and `inc`.
+//
+// When `inc` == 0, it'll be set to math.MaxUint32 which is a secure
+// default.
+//
+// Not safe for concurrent use.
+func Monotonic(entropy io.Reader, inc uint64) io.Reader {
+	if inc == 0 {
+		inc = math.MaxUint32
+	}
+
+	return &monotonic{
+		Reader: bufio.NewReader(entropy),
+		rng:    rand.New(rand.NewSource(time.Now().UnixNano())),
```

Comment:
math/rand is sufficient for this?

Reply:
Isn't this `math/rand`? Or do you mean the global PRNG? I'd like to avoid that big fat lock.